library/std/src/os/cygwin/raw.rs

```diff
@@ -0,0 +1,70 @@
+//! Cygwin-specific raw type definitions
+#![stable(feature = "raw_ext", since = "1.1.0")]
+#![deprecated(
+    since = "1.8.0",
+    note = "these type aliases are no longer supported by \
+            the standard library, the `libc` crate on \
+            crates.io should be used instead for the correct \
+            definitions"
+)]
```

Comment:
Adding these definitions in the past was a mistake, so we shouldn't be adding these to new platforms.

Reply:
`std::os::unix::raw` re-exports some definitions here. I choose to make this crate `pub(crate)` and re-export the `libc` definitions here.